test/framework/modules.py

```diff
@@ -415,8 +419,8 @@ def test_modpath_extensions_for(self):
             # using prepend-path & quoted
             ' prepend-path MODULEPATH "%s/MPI/GCC/4.7.2/OpenMPI/1.6.4"' % mod_dir,
             # conditional 'use' on subdirectory in $HOME, e.g. when --subdir-user-modules is used
-            "if { [ file isdirectory %s/modules/Compiler/GCC/4.7.2 ] } {" % os.environ['HOME'],
-            "    module use %s/modules/Compiler/GCC/4.7.2" % os.environ['HOME'],
+            "if { [ file isdirectory $env(HOME)/modules/Compiler/GCC/4.7.2 ] } {",
+            '    prepend-path MODULEPATH "$env(HOME)/modules/Compiler/GCC/4.7.2"',
```

Comment:
I'm OK with changing the test to dynamically rely on `$HOME`, that's a better test indeed (especially with your `$ARCH` use case in mind), but why change `module use` to `prepend-path`?

Is that required to make the test pass (for `modulecmd.tcl`)? Because this doesn't match with how EasyBuild currently generates modules, it'll use `module use` instead in Tcl modules...